### PR TITLE
GitCommitBear: Fix behaviour for empty body

### PR DESCRIPTION
**Pull request by gurpreetsingh00885**
The bear raised an exception when only the shortlog was included
in the commit message. This commit is meant to fix this bug.

Closes https://github.com/coala/coala-bears/issues/2031

### CODE REVIEW

**Comment by Makman2** [comment]
You addressed an issue marked with `bug`, please use `Fixes` instead of `Closes` now :)

**Comment by gurpreetsingh00885** [comment]
Apart from that what do I need to do to make the test (that failed) pass? And are the changes I made satisfactory?

**Comment by gurpreetsingh00885** [comment]
@Makman2 I need help in getting rid of the AppVeyor build failure.

File: bears/vcs/git/GitCommitBear.py [hunk 1]
```diff
@@ -291,14 +291,15 @@ def check_issue_reference(self, body,
             Whether to enforce presence of issue reference in the body of
             commit message.
         """
+
```
Comment (Makman2) [inline bears/vcs/git/GitCommitBear.py]: Unnecessary change

File: bears/vcs/git/GitCommitBear.py [hunk 2]
```diff
@@ -291,14 +291,15 @@ def check_issue_reference(self, body,
...
         if not body_close_issue:
             return
 
         host = self.get_host_from_remotes()
         if host not in self.SUPPORTED_HOST_KEYWORD_REGEX:
             return
 
-        if body_close_issue_on_last_line:
+        if body_close_issue_on_last_line and body:
```
Comment (Makman2) [inline bears/vcs/git/GitCommitBear.py]: We have to do it differently. Now if the body is empty, but we require a reference in the message, this isn't reported any more.
We need a test case for this so this won't happen again

Comment (gurpreetsingh00885) [reply bears/vcs/git/GitCommitBear.py]: I didn't exactly get what you mean by "...but we require a reference in the message, this isn't reported any more."
What isn't reported anymore?

**Comment by gurpreetsingh00885** [comment]
@Makman2 its here

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
It seems there's a coverage issue: 
```
bears/vcs/git/GitCommitBear.py                                           138      0     70      1    99%   302->304
```

**Comment by Makman2** [comment]
https://github.com/coala/coala-bears/issues/2121 takes over.
